streaming/vision/ade20k.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Callable, Optional, Tuple
+
+import torch
+import torchvision.transforms as T
+import torchvision.transforms.functional as TF
+
+from streaming.vision.base import VisionDataset
+from streaming.vision.transforms import (PadToSize, PhotometricDistoration, RandomCropPair,
+                                         RandomHFlipPair, RandomResizePair)
+
+IMAGENET_CHANNEL_MEAN = (0.485 * 255, 0.456 * 255, 0.406 * 255)
+IMAGENET_CHANNEL_STD = (0.229 * 255, 0.224 * 255, 0.225 * 255)
+
+
+class ADE20k(VisionDataset):
+    """
+    Implementation of the ADE20k dataset using streaming Dataset.
+
+    Args:
+        local (str): Local filesystem directory where dataset is cached during operation.
+        split (str): The dataset split to use, either 'train' or 'val'.
+        remote (str, optional): Remote directory (S3 or local filesystem) where dataset is stored. Default: ``None``.
```

Comment:
< 100 cols (and below)

Reply:
I rearranged it for now manually but we need some automation to check the doc max length. I will look into it in next PR.

Reply:
Is in the isort entry of pyproject.toml